### PR TITLE
An example of adding multiple middlewares.

### PR DESCRIPTION
**Pull request by dvdplm**

### CODE REVIEW

File: examples/multi-middleware.rs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+	fn on_response(&self, started_at: Self::Instant) {
+		println!("[Timings] Response duration {:?}", started_at.elapsed());
+	}
+}
+
+/// Example middleware to keep a watch on the number of total threads started in the system.
+#[derive(Clone)]
+struct ThreadWatcher;
+impl middleware::Middleware for ThreadWatcher {
```
Comment (maciejhirsz) [inline examples/multi-middleware.rs:66]:
```suggestion

impl middleware::Middleware for ThreadWatcher {
```

File: examples/multi-middleware.rs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	type Instant = isize;
```
Comment (maciejhirsz) [inline examples/multi-middleware.rs:67]: Can there be negative threads?

File: examples/Cargo.toml [hunk 3]
```diff
@@ -26,6 +27,10 @@ path = "middleware_ws.rs"
 name = "middleware_http"
 path = "middleware_http.rs"
 
+[[example]]
+name = "multi-middleware"
+path = "multi-middleware.rs"
```
Comment (maciejhirsz) [inline examples/Cargo.toml:32]: All other examples use underscores

```suggestion
name = "multi_middleware"
path = "multi_middleware.rs"
```

**Review by maciejhirsz — APPROVED** [review]

File: examples/multi-middleware.rs (hunk 2 above)
Comment (dvdplm) [reply examples/multi-middleware.rs:68]: Yeah, well, like the subtraction of "end - start" can be negative (these are system threads, so anything can happen while the program executes).

Comment (maciejhirsz) [reply examples/multi-middleware.rs:68]: Yeah, I guess better safe than sorry, and it's just an example

**Pull request merged**